web/ui/react-app/src/pages/rules/Rules.tsx

```diff
@@ -1,15 +1,147 @@
+                <tr>
+                  <td colSpan={3}>
+                    <a href={'#' + g.name}>
+                      <h2>{g.name}</h2>
+                    </a>
+                  </td>
+                  <td>
+                    <h2>{formatRelative(g.lastEvaluation, now())} ago</h2>
+                  </td>
+                  <td>
+                    <h2>{roundUp(parseFloat(g.evaluationTime) * 1000)}ms</h2>
+                  </td>
+                </tr>
+              </thead>
+              <tbody>
+                <tr>
+                  <td className="rules-head">Rule</td>
+                  <td className="rules-head">State</td>
+                  <td className="rules-head">Error</td>
+                  <td className="rules-head">Last Evaluation</td>
+                  <td className="rules-head">Evaluation Time</td>
+                </tr>
+                {g.rules.map((r, i) => {
+                  return (
+                    <tr key={i}>
+                      {r.hasOwnProperty('alerts') ? (
+                        <td>
+                          <span className="rules-head">alert:</span>
+                          <Link
+                            style={{ marginLeft: '3%' }}
+                            to={encodeURI(
+                              `${pathPrefix}/new/graph?g0.expr=ALERTS{alertname="${r.name}"}&g0.tab=1&g0.stacked=0&g0.range_input=1h`
+                            )}
+                          >
+                            {r.name}
+                          </Link>
+                          <br />
+                          <span className="rules-head">expr:</span>
+                          <Link
+                            style={{ marginLeft: '3%' }}
+                            to={`${pathPrefix}/new/graph?g0.expr=${r.query}&g0.tab=1&g0.stacked=0&g0.range_input=1h`}
+                          >
+                            {r.query}
+                          </Link>{' '}
+                          <br />
+                          <Maps map={r.labels} term="labels" />
+                          <Maps map={r.annotations} term="annotations" />
+                        </td>
+                      ) : (
+                        <td>
+                          <span className="rules-head">record:</span> {r.name} <br />
+                          <span className="rules-head">expr:</span>
+                          <Link
+                            style={{ marginLeft: '3%' }}
+                            to={`${pathPrefix}/new/graph?g0.expr=${r.query}&g0.tab=1&g0.stacked=0&g0.range_input=1h`}
+                          >
+                            {r.query}
+                          </Link>
+                        </td>
+                      )}
+                      <td>
+                        <Alert style={{ display: 'inline-table' }}>{r.health.toUpperCase()}</Alert>
+                      </td>
+                      <td>{r.hasOwnProperty('lastError') ? r.lastError : null}</td>
```

Comment:
```suggestion
                      <td>{r.lastError}</td>
```
If it hasn't own property it will not render anyway